### PR TITLE
feat: add log to show the version status of cli

### PR DESCRIPTION
**Pull request by AayushSaini101**
Resolves: #1371

### CODE REVIEW

**Comment by Shurtu-gal** [comment]
@AayushSaini101 I suppose it the message should be infrequent so as not to bother users.

**Comment by AayushSaini101** [comment]
> @AayushSaini101 I suppose it the message should be infrequent so as not to bother users.

@Shurtu-gal Thanks, used the probaility method to show the message when the probaility is less than 0.5. 

Sample Testing of message: 

Screenshot 2024-04-27 at 10 05 09 AM

File: src/utils/versionVerification.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  const packageJson = JSON.parse(await fs.readFile('package.json', 'utf-8') as string);
+  const currentVersion = `v${packageJson.version}`;
+  // Compare the versions
+  if (latestVersion !== currentVersion) {
+    // Avoid sending message frequently
+    const randomProbability = Math.random();
+    if (randomProbability < 0.5) {return `There is a newer version available: ${latestVersion}. You are currently using ${currentVersion}.`;}
+  }
+}
+checkLatestVersion();
```
Comment (Shurtu-gal) [inline src/utils/versionVerification.ts:27]: Is there any reason the function is being called here?

Comment (AayushSaini101) [reply src/utils/versionVerification.ts]: thanks @Shurtu-gal no used, I was testing some changes removed

Comment (AayushSaini101) [reply src/utils/versionVerification.ts]: @Shurtu-gal If we are doing infrequent messages, how can we test this 🤔

**Comment by Amzani** [comment]
> @AayushSaini101 I suppose it the message should be infrequent so as not to bother users. 
Agree, especially if we don't have (yet) an auto-update feature.

**Comment by Amzani** [comment]
@AayushSaini101 why not just use https://github.com/oclif/plugin-warn-if-update-available, instead of having the random check for simplicity the plugin provides a periodic check that is more common. 

We can check every 24h for instance.

Some other useful plugins we can add:
- https://github.com/oclif/plugin-version
- https://github.com/oclif/plugin-update

**Comment by AayushSaini101** [comment]
> @AayushSaini101 why not just use https://github.com/oclif/plugin-warn-if-update-available, instead of having the random check for simplicity the plugin provides a periodic check that is more common.
> 
> We can check every 24h for instance.
> 
> Some other useful plugins we can add:
> 
> * https://github.com/oclif/plugin-version
> * https://github.com/oclif/plugin-update
Thanks @Amzani I am exploring them

**Review by KhudaDad414 — COMMENTED** [review]
regarding the message that we show, I am seeing the following format in a few places including `Next.js`, `Storybook`, `Pnpm`. 

Screenshot 2024-04-29 at 11 53 10

**Comment by AayushSaini101** [comment]
> https://github.com/oclif/plugin-warn-if-update-available

@Amzani @KhudaDad414 Added plugin and configuration to run in 24 hour.

**Comment by Shurtu-gal** [comment]
Great work @AayushSaini101.

**Review by KhudaDad414 — APPROVED** [review]
LGTM

**Review by Shurtu-gal — APPROVED** [review]
LGTM 🚀

**Comment by KhudaDad414** [comment]
@AayushSaini101  the tests seem to be cancelled. can you push an empty commit to re-run the tests?

**Review by peter-rr — COMMENTED** [review]
LGTM 🚀

**Pull request merged**